### PR TITLE
[BREAKING] fix(Subscribe): Add option to subscribe with holes in prefixes.

### PR DESCRIPTION
**Pull request by ahsanbarkati**
Add an argument in `(*DB) Subscribe()` for the ignore string so that the subscriber can subscribe to a prefix with holes (or ignore bytes).

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1658)

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -1828,13 +1828,14 @@ type KVList = pb.KVList
 // This function blocks until the given context is done or an error occurs.
 // The given function will be called with a new KVList containing the modified keys and the
 // corresponding values.
-func (db *DB) Subscribe(ctx context.Context, cb func(kv *KVList) error, prefixes ...[]byte) error {
+func (db *DB) Subscribe(ctx context.Context, cb func(kv *KVList) error, ignore string,
```
Comment (NamanJain8) [inline db.go:1831]: Update the function description. Rest looks good. 🎉

**Review by NamanJain8 — APPROVED** [review]
Just a small comment. Rest looks good.

File: db.go (hunk 1 above)
Comment (ahsanbarkati) [reply db.go:1833]: Done. :)

File: db.go [hunk 2]
```diff
@@ -1822,19 +1822,22 @@ func (db *DB) BannedNamespaces() []uint64 {
...
+// Subscribe can be used to watch key changes for the given key prefixes and the ignore string.
 // At least one prefix should be passed, or an error will be returned.
 // You can use an empty prefix to monitor all changes to the DB.
+// Ignore string is the byte ranges for which prefix matching will be ignored.
+// For example: ignore = "2-3", and prefix = "abcde" will match for keys "abxxe", "abdfe" etc.
 // This function blocks until the given context is done or an error occurs.
 // The given function will be called with a new KVList containing the modified keys and the
 // corresponding values.
-func (db *DB) Subscribe(ctx context.Context, cb func(kv *KVList) error, prefixes ...[]byte) error {
+func (db *DB) Subscribe(ctx context.Context, cb func(kv *KVList) error, ignore string,
```
Comment (jarifibrahim) [inline db.go:1833]: Can't we use the pb.Match here directly? There is a list of prefixes. What if I want to specify multiple ignore strings? Right now you're using the same one for all prefixes.

Also, this is a breaking change (Public API change)

Comment (ahsanbarkati) [reply db.go]: Yeah. That would be better to have. Thanks

**Review by jarifibrahim — APPROVED** [review]

**Pull request merged**